app/components/Map.client.tsx

```diff
@@ -12,43 +14,89 @@ interface Props {
 }
 
 const Map = ({ map, setMap, setMapLoaded }: Props) => {
+  const mapContainerRef = useRef<HTMLDivElement>(null);
+
   useEffect(() => {
-    if (!setMap) return;
+    if (!setMap || !mapContainerRef.current) return;
 
     const bounds = new maplibregl.LngLatBounds(
       new maplibregl.LngLat(-82.01409567385569, 30.679059125170696),
-      new maplibregl.LngLat(-80.92207334522604, 32.11595891326837),
+      new maplibregl.LngLat(-80.92207334522604, 32.11595891326837)
     );
 
     const _map = new maplibregl.Map({
-      container: "mapContainer",
-      // @ts-ignore We are loading the style as JSON.
-      style,
+      container: mapContainerRef.current,
+      style: style as StyleSpecification,
       center: [-81.40348956381558, 31.41113196761974],
       zoom: 9,
     });
 
     _map.fitBounds(bounds);
 
-    setMap(_map);
+    _map.on("load", () => {
+      addMapLayers(_map);
+      setMap(_map);
+      setMapLoaded(true);
+    });
 
     return () => {
       if (_map) _map.remove();
       setMap(undefined);
       setMapLoaded(false);
     };
-  }, [setMap]);
+  }, [setMap, setMapLoaded]);
 
-  useEffect(() => {
-    if (!map) return;
-    map.on("load", async () => {
-      setMapLoaded(true);
+  const addMapLayers = (_map: maplibregl.Map) => {
+    // Add all the layers here
+    // Google Satellite
+    _map.addSource('google-satellite', {
```

Comment:
We should add an array of maplayers to the [config file](https://github.com/ecds/georgia-coast-atlas-frontend/blob/dev/app/config.ts) and loop through them - sort of like the islands. It would be an array of objects with keys like "id" and "label", eg:

~~~javascript
// app/config.ts
export const mapLayers = [
  {
    id: "usgs-topo",
    label: "USGS Topo" ,
    tiles: [...],
    attribution: "© USGS"
  },
  ....
]
~~~